pulp_rpm/app/modulemd.py

```diff
@@ -1,69 +1,80 @@
 import json
+import os
 import tempfile
 
-from django.db.utils import IntegrityError
 from pulpcore.app.models.content import Artifact
-from pulp_rpm.app.constants import PULP_MODULE_ATTR
+from pulp_rpm.app.constants import PULP_MODULE_ATTR, PULP_MODULEDEFAULTS_ATTR
 
 import gi
 gi.require_version('Modulemd', '2.0')
 from gi.repository import Modulemd as mmdlib  # noqa: E402
 
 
 def _create_snippet(snippet_string):
-    """Create snippet of modulemd[-defaults] as artifact."""
-    tmp_file = tempfile.NamedTemporaryFile()
-    with open(tmp_file.name, 'w') as f:
-        f.write(snippet_string)
-    artifact = Artifact.init_and_validate(tmp_file.name)
-    try:
-        artifact.save()
-    except IntegrityError:
-        artifact = Artifact.objects.get(sha256=artifact.sha256)
-    return artifact
+    """
+    Create snippet of modulemd[-defaults] as artifact.
 
+    Attributes:
```

Comment:
s/Attributes/Args/
This is a function which has arguments. A class can have attributes.
For more examples see https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

Reply:
very good resource thanks